packages/protocol/src/peer/ControllerCommissioningFlow.ts

```diff
@@ -1225,11 +1312,18 @@ export class ControllerCommissioningFlow {
             reArmFailsafeInterval.start();
         }
 
-        const transitionResult = await this.transitionToCase(
-            this.interactionClient.address,
-            // Assume concurrent connections are supported if not know (which should not be the case when we came here)
-            isConcurrentFlow,
-        );
+        let transitionResult: InteractionClient | undefined;
+        try {
+            transitionResult = await this.transitionToCase(
+                this.interactionClient.address,
+                // Assume concurrent connections are supported if not know (which should not be the case when we came here)
+                isConcurrentFlow,
+            );
+        } catch (error) {
+            const commError = repackErrorAs(error, CommissioningError);
```

Comment:
Changing the error class potentially drops diagnostic information:

* original error class name/support for typeof
* extra fields on SuppressedError and AggregateError
* extra fields on non-standard Error extensions
* current stack trace (which is likely truncated from the original error, or possibly even different in the case of promises)

I think we also don't copy the new Error#cause property, although that's a quick fix

So maybe we should be throwing the new error with the original error as the cause? Except maybe in limited circumstances where we receive a specific error and know the stack doesn't provide additional useful details

Reply:
I adjusted it in this pace and also adjusted "repack" method also when not used here now because of the second parameter. Please have a look again